File: Javascript/FCC/insertionSort.js
const insertionSort = (arr) => {
  for (let k = 0; k < arr.length; k++) {
    for (let i = 0; i < arr.length - 1; i++) {
      if (arr[i] > arr[i + 1]) {
        let temp = arr[i];
        arr[i] = arr[i + 1];
        arr[i + 1] = temp;
      }
    }
  }
  console.log(arr);
};

// insertionSort([1, 4, 2, 8, 345, 123]);
insertionSort([7, 4, 2, 8, 345, 1, 123]);

File: Python/geeksforgeeks/insertionsort.py
def insertionsort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1

        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key


if __name__ == '__main__':
    arr = [23, 32, 2, 100, 52, 10]
    insertionsort(arr)
    print(arr)
